### ISSUE TITLE
silhouette score

### ISSUE THREAD

**Issue by terraswat** [ucscHexmap/hexagram#19]
Improve our clumpiness/density stats. A silhouette score may do this or at least part of if.

A measure of "clumpiness" of groups of samples, where the groups could be based on attributes or on user-selection. The best way to do that is to utilize what's called a Silhouette score. Essentially, it computes how well given data points (samples) fit into the specified grouping. There's a python library that does all the computations (http://scikit-learn.org/stable/modules/generated/sklearn.metrics.silhouette_score.html). I will have to produce a full matrix of Euclidean distances in the map points but other than that this function only takes group assignments.

I imagine it would be a new button or a menu item in the tool. Upon clicking on it a popup will display asking the user to select the groups(s) via a drop down, which will contain all the categorical and binary attributes as well as user selections. Then, it will do the computation on the server side and return a value (average silhouette) for the grouping(s). Maybe it will display it as a popup or in some other way.

**Comment by mrJeppard** [comment]
lees L spatial autocorrelation metric is being used to calculate density. It is effective
